_posts/en/2021-06-08-working-in-the-open-part-2.md

```diff
@@ -0,0 +1,147 @@
+- [Web Experience Toolkit](https://www.canada.ca/en/treasury-board-secretariat/services/government-communications/web-experience-toolkit.html){:target="_blank"} is a Government of Canada initiatives that "includes reusable components for building and maintaining innovative websites that are accessible, usable, and interoperable.
+  These reusable components are open source software and free for use by departments and external Web communities."
+  It is a great example of how a team was able to work in the open to benefit the government as a whole but also the society itself.
+- [GC Service & Digital Target Enterprise Architecture](https://wiki.gccollab.ca/index.php?title=GC_Service_%26_Digital_Target_Enterprise_Architecture&mobileaction=toggle_view_desktop){:target="_target"}
+- [1988 National Bureau of Standards' Management Guide to
+Software Reuse](https://www.govinfo.gov/content/pkg/GOVPUB-C13-dca04f7b7fc21321e39de88b35b2f1d3/pdf/GOVPUB-C13-dca04f7b7fc21321e39de88b35b2f1d3.pdf) (PDF)
+
+Needless to say that public administrations in general are ripe for reusing solutions and ideas to increase the quality of their services to citizens.
+
+### Manage secrets
+
+In software development, certain parts of the code may be a corporate secret, such as a private key to a resource, passwords, etc.
+
+These must never be included in the public source code itself but rather be managed separately.
+
+Since, software source code is nothing more than structured text that can be understood by a computer, the same concept of pulling out secrets parts may be taken with other types of documents.
+
+This means that when a small part of your work must not be shared publicly, you can exclude it and store it in a secure location instead of locking down the entire document.
+
+This way, the material that is unclassified can still be made available publicly and may be collectively worked on while the parts that are to be secured can be edited separately.
+
+By default, you should strive to keep as many of your working documents in the open section of your workspace.
+But when needed, you can save sensitive information in the private folders and keep working in the open for the rest.
+
+### Manage leaks
+
+In software development, we expect that there will be a security issue at some point, while we continue to strive to minimize that eventuality.
+
+By putting in place a remediation process, we ensure that even if there was to be a leak, appropriate mesures are in place to address it as quickly as possible.
+And we make sure that everyone knows how to run through the process, if it's not fully automated.
```

Comment:
I feel like we need a bit more here....since this article is talking a lot about docs rather than code, what would the remediation process look like for open documents?  Are we suggesting folks have something in place beyond existing departmental processes? Are we suggesting that they know and understand what these are (and that they should know and understand them even if not working openly as accidental leaks happen regardless).

Reply:
Hmm, good point... Well, it kind of depends what we're working on I would say. In software development, if you have keys that have been exposed, you need a process in place to first find that but also to pinpoint any use of those keys, have them revoked and replaced by new ones.

In a world of documents, the use of secret keys is not as much plausible... Passwords? If we're talking about sensitive information leaked by mishap, then internal processes for divulgation should most likely be known.

Might need a brainstorm on that... I guess by default we only use unclassified data in our work so that type of question is not something we tend to need to address that often.

Reply:
I tried to better explain how to manage this in a non-development situation.

Reply:
I'm still unsure about this section. I wonder if we actually need it since the section above says to separate out secrets/private personal info.  If that is done there should be nothing that would get leaked?  But if someone did include personal info in an open file (and so leak it), there is an already defined process that needs to be followed. At ESDC this is guided by the 'Privacy Breach Directive' that talks about containment, preliminary assessment, notifying senior mgmt and security.

Reply:
I'll simply note the Directive itself then! In IT security, you have to take all measures possible to minimize the risks. But you also need to plan in a way that if the leak actually happens, you know exactly what to do :smile: